code/game/objects/effects/anomalies.dm

```diff
@@ -183,3 +183,30 @@
 	if( T && istype(T,/turf/simulated) && prob(turf_removal_chance) )
 		T.ex_act(ex_act_force)
 	return
+
+/////////////////////
+
+/obj/effect/anomaly/atmos
+	name = "transformative gas anomaly"
+	icon_state = "electricity2"
+	var/gas_type
+
+/obj/effect/anomaly/atmos/New()
+	..()
+	gas_type = pick(GAS_N2O, GAS_CO2, GAS_N2)
```

Comment:
I think this can be made more efficient if you instead assign `gas_type` to one of `LINDA_SPAWN_CO2`, `LINDA_SPAWN_N2O` or `LINDA_SPAWN_NITROGEN`.

And then in your `anomalyEffect()`, you can remove the switch statement and just call:
```dm
T.atmos_spawn_air(LINDA_SPAWN_20C | gas_type, 10)
```

Reply:
I think the performance impact is minimal - and the code is clearer this way. Plus, he'll have to use the gas type later on.